### DISCUSSION TITLE
Read/write transactions?

### DISCUSSION THREAD

**Discussion by timuster** [supabase/supabase#2154] [Questions] [answered]
Does Supabase support read/write transactions like [Firebase transactions]( https://firebase.google.com/docs/firestore/manage-data/transactions#transactions)?

In other words, does Supabase provide a way to update a row (specifically a JSONB column) based on its current value (and avoid race conditions)?

**Comment by steve-chavez** [comment, marked as answer]
Not for now, but it's a planned feature. Check this previous discussion: https://github.com/supabase/supabase/discussions/526.

> In other words, does Supabase provide a way to update a row (specifically a JSONB column) based on its current value (and avoid race conditions)?

You can do that logic in a SQL function and call it through [supabase.rpc()](https://supabase.io/docs/reference/javascript/rpc).

Check this example for incrementing a table number: https://github.com/supabase/supabase/discussions/909#discussioncomment-516407
